### PR TITLE
Default scan active

### PR DESCRIPTION
**Pull request by ajayparida**

### CODE REVIEW

File: samples/wifi/scan/src/main.c [hunk 1]
```diff
@@ -182,13 +182,28 @@ static int wifi_scan(void)
...
+	if (net_mgmt(NET_REQUEST_WIFI_SCAN, iface, &params,
+		     sizeof(struct wifi_scan_params))) {
+		LOG_ERR("Passive scan request failed");
+
+		return -ENOEXEC;
+	}
+
+	printk("Passive scan requested\n");
+#else
 	if (net_mgmt(NET_REQUEST_WIFI_SCAN, iface, NULL, 0)) {
```
[the remark this answers is not in the document]
Comment (ajayparida) [reply samples/wifi/scan/src/main.c:199]: Added Kconfig option. Please review

File: samples/wifi/scan/Kconfig [hunk 2]
```diff
@@ -15,4 +15,23 @@ config WIFI_MAC_ADDRESS
...
 		use the default MAC address (locally administrative). This is only used
 		when the nRF7002 OTP is not programmed with a MAC address.
+
+choice  WIFI_SCAN_TYPE_SELECT
+	prompt "Scan type"
+	default WIFI_SCAN_TYPE_ACTIVE
+	help
+		WiFi scan type to be used to perform current scan operation.
+		If this is not set default action to do active scan.
+		This scan type can be overridden by os level setting.
```
Comment (krish2718) [inline samples/wifi/scan/Kconfig:25]:
```suggestion
		WiFi scan type to be used to perform current scan operation.
		If this is not set default action is to do active scan.
```

File: samples/wifi/scan/prj.conf [hunk 3]
```diff
@@ -43,3 +43,7 @@ CONFIG_NET_MGMT_EVENT_QUEUE_TIMEOUT=5000
 # Raw scan Options
 CONFIG_WIFI_MGMT_RAW_SCAN_RESULTS=n
 CONFIG_WIFI_MGMT_RAW_SCAN_RESULTS_ONLY=n
+
+# Below configs need to be modified to set scan type
+#CONFIG_WIFI_SCAN_TYPE_ACTIVE=y
+#CONFIG_WIFI_SCAN_TYPE_PASSIVE=y
```
Comment (krish2718) [inline samples/wifi/scan/prj.conf:49]: Please remove these commented flags

File: samples/wifi/scan/src/main.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (krish2718) [reply samples/wifi/scan/src/main.c:199]: Why to duplicate entire code? Always sends `params` just change `params.scan_type` according to the input and other conditions.

**Review by krish2718 — CHANGES_REQUESTED** [review]

**Review by rado17 — APPROVED** [review]

File: samples/wifi/scan/src/main.c [hunk 4]
```diff
@@ -182,7 +182,15 @@ static int wifi_scan(void)
 {
 	struct net_if *iface = net_if_get_default();
 
-	if (net_mgmt(NET_REQUEST_WIFI_SCAN, iface, NULL, 0)) {
+	struct wifi_scan_params params = {0};
+
+	params.scan_type = WIFI_SCAN_TYPE_ACTIVE;
```
Comment (krish2718) [inline samples/wifi/scan/src/main.c:187]: This could be in the else below.

**Review by krish2718 — APPROVED** [review]

**Review by bama-nordic — APPROVED** [review]

File: samples/wifi/scan/README.rst [hunk 5]
```diff
@@ -20,9 +20,21 @@ Overview
 ********
 
 This sample can perform Wi-Fi scan operations in the 2.4GHz and 5GHz bands.
+Default scan type is active scan. Optionaly user can select scan type (active/passive).
```
Comment (kittydepa) [inline samples/wifi/scan/README.rst:23]:
```suggestion
The default scan type is active scan. However, you can configure the scan type to be either active or passive.
```

File: samples/wifi/scan/README.rst [hunk 6]
```diff
@@ -20,9 +20,21 @@ Overview
...
 Using this sample, the development kit can scan for available access points in :abbr:`STA (Station)` mode.
 
+|config|
+
+You may configure the Wi-Fi scan type in the :file:`prj.conf` file:
+
+* Scan type
+
+.. note::
+       You can also use ``menuconfig`` to enable ``Scan type`` option.
```
Comment (kittydepa) [inline samples/wifi/scan/README.rst:34]:
```suggestion
       You can also use ``menuconfig`` to enable the ``Scan type`` option.
```

**Review by kittydepa — CHANGES_REQUESTED** [review]

File: samples/wifi/scan/README.rst [hunk 7]
```diff
@@ -20,9 +20,21 @@ Overview
...
 This sample can perform Wi-Fi scan operations in the 2.4GHz and 5GHz bands.
+The default scan type is active scan. However, you can configure the scan type to be either active or passive.
 
 Using this sample, the development kit can scan for available access points in :abbr:`STA (Station)` mode.
 
+|config|
+
+You may configure the Wi-Fi scan type in the :file:`prj.conf` file:
+
+* Scan type
```
Comment (peknis) [inline samples/wifi/scan/README.rst:31]: Do we need this line?

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/wifi/scan/README.rst [hunk 8]
```diff
@@ -20,9 +20,19 @@ Overview
 ********
 
 This sample can perform Wi-Fi scan operations in the 2.4GHz and 5GHz bands.
+The default scan type is active scan. However, you can configure the scan type to be either active or passive.
```
Comment (peknis) [inline samples/wifi/scan/README.rst:23]:
```suggestion
The default scan type is active scan.
However, you can configure the scan type to be either active or passive.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/wifi/scan/Kconfig [hunk 9]
```diff
@@ -15,4 +15,22 @@ config WIFI_MAC_ADDRESS
 		WiFi MAC address to be used by the sample. If not set, the sample will
 		use the default MAC address (locally administrative). This is only used
 		when the nRF7002 OTP is not programmed with a MAC address.
+
+choice  WIFI_SCAN_TYPE_SELECT
+	prompt "Scan type"
+	default WIFI_SCAN_TYPE_ACTIVE
+	help
+		WiFi scan type to be used to perform current scan operation.
```
Comment (peknis) [inline samples/wifi/scan/Kconfig:23]:
```suggestion
		Wi-Fi scan type to be used to perform current scan operation.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/wifi/scan/README.rst (hunk 7 above)
Comment (ajayparida) [reply samples/wifi/scan/README.rst:30]: No removed this line, as refernce in prj.conf is no more. Updated doc accordingly.

File: samples/wifi/scan/README.rst [hunk 10]
```diff
@@ -20,9 +20,20 @@ Overview
 ********
 
 This sample can perform Wi-Fi scan operations in the 2.4GHz and 5GHz bands.
+The default scan type is active scan.
+However, you can configure the scan type to be either active or passive.
 
 Using this sample, the development kit can scan for available access points in :abbr:`STA (Station)` mode.
```
Comment (peknis) [inline samples/wifi/scan/README.rst:27]: You might want to add here the Configuration heading. See the sample doc template: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/templates/sample_README.html#configuration

Comment (ajayparida) [reply samples/wifi/scan/README.rst:27]: Yeah, added, please review

Comment (peknis) [reply samples/wifi/scan/README.rst:27]: Not quite yet there. Proposing to have:

Configuration
***************
|config|

note about the scan type

Configuration options
==============

I believe :kconfig:option:`CONFIG_WIFI_SCAN_TYPE_SELECT` should work here, as the new option is going to be available in the Kconfig reference, right?

Comment (peknis) [reply samples/wifi/scan/README.rst:27]: The parsing went a bit wrong in my comment, but I hope you get the idea. Just like in the template.

Comment (peknis) [reply samples/wifi/scan/README.rst:27]: Sorry for stealing the review from you @kittydepa. Please continue from here.

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/wifi/scan/README.rst [hunk 11]
```diff
@@ -20,9 +20,30 @@ Overview
...
 
+Configuration
+*************
+
+|config|
+
+Select the scan type to be used by the sample:
```
Comment (peknis) [inline samples/wifi/scan/README.rst:33]:
```suggestion
Select the scan type to be used by the sample.
```
Alternatively:
Now, if there are no other important configuration options, you could dump the Configuration options section and just have it in this Configuration section.
Something like: "Use the :kconfig:option:`CONFIG_WIFI_SCAN_TYPE_SELECT` Kconfig option to set the scan type to be used by the sample."

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by kittydepa — APPROVED** [review]

**Pull request merged**